components/genome-browser-view.js

```diff
@@ -0,0 +1,146 @@
+    // if there are more filtered files than we want to display on one page, we will paginate
+    const browserTotalPages = Math.ceil(
+      filteredFiles.length / NUMBER_OF_FILES_FOR_DISPLAY
+    );
+    setTotalPage(browserTotalPages);
+    setCurrentPage(1);
+    const includedFiles = filteredFiles.slice(0, NUMBER_OF_FILES_FOR_DISPLAY);
+    setDisplayedFiles(includedFiles);
+  }
+
+  function handlePagination(pageDirection) {
+    if (pageDirection === "plus") {
+      const startIdx = currentPage * NUMBER_OF_FILES_FOR_DISPLAY;
+      const endIdx = (currentPage + 1) * NUMBER_OF_FILES_FOR_DISPLAY;
+      const displayedFiles = filteredFiles.slice(startIdx, endIdx);
+      setDisplayedFiles(displayedFiles);
+      setCurrentPage(currentPage + 1);
+    } else {
+      const startIdx = (currentPage - 2) * NUMBER_OF_FILES_FOR_DISPLAY;
+      const endIdx = (currentPage - 1) * NUMBER_OF_FILES_FOR_DISPLAY;
+      const displayedFiles = filteredFiles.slice(startIdx, endIdx);
+      setDisplayedFiles(displayedFiles);
+      setCurrentPage(currentPage - 1);
+    }
+  }
+
+  return (
+    showBrowserData && (
+      <>
+        {files.length > 0 ? (
+          <>
+            <DataAreaTitle>Genome Browser</DataAreaTitle>
+            <DataPanel>
+              <DataArea>
+                <DataItemLabel>Number of tracks:</DataItemLabel>
+                <DataItemValue>{filteredFiles.length}</DataItemValue>
+              </DataArea>
+              <GenomeBrowserFacets
+                files={files}
+                handleFacetList={handleFacetList}
+                filteredFiles={filteredFiles}
+                selectedFilters={selectedFilters}
+              />
+              <GenomeBrowser
+                files={displayedFiles}
+                assembly={assembly}
+                coordinates={coordinates}
+              />
+              {totalPage > 1 && (
+                <div className="text-center mt-6">
+                  <div>
+                    <button
+                      disabled={currentPage === 1}
+                      onClick={() => handlePagination("minus")}
+                    >
+                      <ChevronLeftIcon
+                        className={`h-8 w-8 ${
+                          currentPage === 1
+                            ? "text-gray-200"
+                            : "hover:bg-gray-200"
+                        }`}
+                      />
+                    </button>
+                    <button
+                      disabled={currentPage === totalPage}
+                      onClick={() => handlePagination("plus")}
+                    >
+                      <ChevronRightIcon
+                        className={`h-8 w-8 ${
+                          currentPage === totalPage
+                            ? "text-gray-200"
+                            : "hover:bg-gray-200"
+                        }`}
+                      />
+                    </button>
+                  </div>
```

Comment:
I think it’d be nice to make these buttons React components themselves to help document this render, and make it smaller. Maybe even the whole pager mechanism in its own component.